geowebcache/core/src/main/java/org/geowebcache/locks/MemoryLockProvider.java

```diff
@@ -14,12 +14,12 @@
  */
 package org.geowebcache.locks;
 
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.locks.ReentrantLock;
 import java.util.logging.Level;
 import java.util.logging.Logger;
-import org.apache.commons.codec.digest.DigestUtils;
 import org.geotools.util.logging.Logging;
-import org.geowebcache.GeoWebCacheException;
 
 /**
  * An in memory lock provider based on a striped lock
```

Comment:
Please update this javadoc with a summary of how this class works now (some details about lack of bouds, how the map is not actually growing to use infinite memory, and the nested locking).